### PR TITLE
feat(asset-api): use the new Google cloud function that exercises the asset inventory api

### PR DESCRIPTION
**Pull request by obs-gh-colinhutchinson**
## What does this PR do?

This PR introduces significant changes to our existing Terraform configuration. It revolves around transitioning from using generic GCP APIs to using the asset inventory API for enhanced capabilities and performance.

## Motivation

Avoid the cloud function timeout and avoid the GCP API quota limits

## Testing

Tested locally with a new observe account and new GCP account
```
module "observe_gcp_collection" {
  source   = "github.com/observeinc/terraform-google-collection?ref=feat%2FOB-19512-export-and-feed"
  #source = "../../../google-collection/"
  name     = format(var.name_format, "env")
  resource = "projects/${var.project_id}"
}
```

refs: OB-19512
related: https://github.com/observeinc/google-cloud-functions/pull/11

### CODE REVIEW

File: function.tf [hunk 1]
```diff
@@ -83,16 +155,20 @@ resource "google_cloudfunctions_function_iam_member" "cloud_scheduler" {
...
 
   http_target {
     http_method = "POST"
     uri         = google_cloudfunctions_function.this[0].https_trigger_url
 
+    headers = {
+      Content-Type = "application/json"
+    }
+
+    body        = base64encode("{}")
```
Comment (bendrucker) [inline function.tf:170]: Having to send an empty JSON object is unexpected, why is this needed?

Comment (obs-gh-colinhutchinson) [reply function.tf:170]: At one point I was sending GCP resource types, and asset types as arguments because I thought customer might want to disable / enable. Turns out there's no costs / downsides to doing everything but the result is (currently) the Python code is expecting a body that is json

**Review by joey-skittlz — APPROVED** [review]

**Review by arthur-observe — APPROVED** [review]

File: examples/basic_usage/main.tf
Comment (bendrucker) [inline examples/basic_usage/main.tf:1]: `examples/basic` is typically the first example I make in any project where there's an `examples/` dir. I'll often do `testdata/basic` in Go too when I know there's multiple testdata cases and I'm building the initial one.

Every example is a usage example, so including `usage` in example names is superfluous.

File: examples/basic_usage/main.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module "observe_collection_example" {
```
Comment (bendrucker) [inline examples/basic_usage/main.tf:1]: Examples should be able to form the initial basis of a user's configuration. Avoid putting `example` in any resource names. Name things the way you want end users to name them. `module.observe_collection` or `module.observe_google_collection` would be good choices.

File: examples/basic_usage/main.tf [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  source = "../../"
```
Comment (bendrucker) [inline examples/basic_usage/main.tf:2]: Repeating for emphasis, this means the examples are broken on the Terraform Registry and cannot be copied. 

https://developer.hashicorp.com/terraform/language/modules/develop/structure

> Because examples will often be copied into other repositories for customization, any module blocks should have their source set to the address an external caller would use, not to a relative path.

To use examples as tests against HEAD, rewrite that external caller address to this relative path.

File: function.tf [hunk 4]
```diff
@@ -38,10 +38,23 @@ resource "google_pubsub_topic_iam_member" "cloudfunction_pubsub" {
   member = "serviceAccount:${google_service_account.cloudfunction[0].email}"
 }
 
+resource "google_storage_bucket" "this" {
+  name     = "${var.name}-bucket"
+  location = "US"
+
+  force_destroy = true
```
Comment (bendrucker) [inline function.tf:45]: This is typically not a good default when you're making durable datastores for users. What potentially makes this an exception is the fact that the bucket is essentially just a temporary buffer for the exported data.

_This_ would be a good thing to clarify in a code comment, e.g.

> Since the bucket is just a temporary storage for asset export objects until the function can process them, we want to implicitly delete any leftover objects if Terraform plans to remove the bucket

File: function.tf [hunk 5]
```diff
@@ -61,11 +75,52 @@ resource "google_cloudfunctions_function" "this" {
...
+    }
+  }
+
+  available_memory_mb = var.function_available_memory_mb
+  timeout             = var.function_timeout
+  max_instances       = var.function_max_instances
+
+  source_archive_bucket = var.function_bucket
+  source_archive_object = var.function_object
+  entry_point           = "gcs_to_pubsub"  # This should be the entry point of your Python function for GCS bucket events
```
Comment (bendrucker) [inline function.tf:112]: Unclear who this comment is for

File: variables.tf [hunk 6]
```diff
@@ -125,13 +129,13 @@ variable "folder_include_children" {
 variable "function_bucket" {
   description = "GCS bucket containing the Cloud Function source code"
   type        = string
-  default     = "observeinc"
+  default     = "observeinc-colin"
```
Comment (bendrucker) [inline variables.tf:132]: This was not ready for release!

File: variables.tf [hunk 7]
```diff
@@ -125,13 +129,13 @@ variable "folder_include_children" {
...
 }
 
 variable "function_object" {
   description = "GCS object key of the Cloud Function source code zip file"
   type        = string
-  default     = "google-cloud-functions-v0.2.0.zip"
+  default     = "google-cloud-functions-v0.3.0-alpha.7.zip"
```
[the remark this answers is not in the document]
Comment (bendrucker) [reply variables.tf:138]: This was not ready for release!

File: variables.tf [hunk 8]
```diff
@@ -169,6 +173,18 @@ variable "function_disable_logging" {
   default     = false
 }
 
+variable "function_output_bucket" {
+  description = "The Google Cloud Storage (GCS) bucket where the function output will be stored."
+  type        = string
+  default     = "chutchinson-export-assets"
```
Comment (bendrucker) [inline variables.tf:179]: Also this
